pulsar/log/logger.go

```diff
@@ -0,0 +1,57 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package log
+
+// Logger and Entry interfaces are inspired by sirupsen/logrus
+
+// Fields type, used to pass to `WithFields`.
+type Fields map[string]interface{}
+
+// Logger describes the interface that must be implemeted by all loggers.
+type Logger interface {
+	SubLogger(fields Fields) Logger
+
+	WithFields(fields Fields) Entry
+	WithField(name string, value interface{}) Entry
```

Comment:
IMO there should be also `WithError(err) Entry` method, `WithField("cause", err)`  is not 1:1 replacement and more error-prone (e.g. somebody can make a typo field name).

Reply:
Make sense, it will be handy and less error-prone if the `WithError` is available for pulsar-client-go, although the logging features are already complete.